packages/gatsby/src/internal-plugins/partytown/gatsby-browser.tsx

```diff
@@ -1,64 +1,24 @@
-import React, { ReactElement, useState } from "react"
-import type { GatsbySSR } from "gatsby"
-import { Partytown } from "@builder.io/partytown/react"
-import { PartytownContext, ScriptProps } from "gatsby-script"
+import { collectedScriptsByPage } from "gatsby-script"
+import { injectPartytownSnippet } from "./utils/inject-partytown-snippet"
+import type { GatsbyBrowser } from "gatsby"
 
-interface ICollectedForwardsState {
-  collectedForwards: Set<string>
-  collectedAnyScript: boolean
-}
-
-function PartytownProvider({ children }): ReactElement {
-  const [{ collectedForwards, collectedAnyScript }, setState] =
-    useState<ICollectedForwardsState>({
-      collectedForwards: new Set(),
-      collectedAnyScript: false,
-    })
+export const onInitialClientRender: GatsbyBrowser[`onInitialClientRender`] =
+  (): void => {
+    injectPartytownSnippet(window.location.pathname)
+    collectedScriptsByPage.delete(window.location.pathname)
+  }
 
-  return (
-    <PartytownContext.Provider
-      value={{
-        collectScript: (newScript: ScriptProps): void => {
-          let stateShouldChange = false
-          const potentialNewState = {
-            collectedAnyScript,
-            collectedForwards,
-          }
+export const onRouteUpdate: GatsbyBrowser[`onRouteUpdate`] = ({
+  prevLocation,
+  location,
+}): void => {
+  if (!prevLocation) {
+    return
+  }
 
-          if (!collectedAnyScript) {
-            potentialNewState.collectedAnyScript = true
-            stateShouldChange = true
-          }
+  collectedScriptsByPage.delete(window.location.pathname)
```

Comment:
Use react state instead to make sure you have the correct path.
```suggestion
  collectedScriptsByPage.delete(location.pathname)
```